code/modules/mob/new_player/new_player.dm

```diff
@@ -297,22 +287,71 @@
 				to_chat(src, "<span class='notice'>Vote successful.</span>")
 
 
+/mob/new_player/proc/make_me_an_observer()
+	if(!SSticker || SSticker.current_state == GAME_STATE_STARTUP)
+		to_chat(src, "<span class='warning'>The game is still setting up, please try again later.</span>")
+		return FALSE
+
+	if(QDELETED(src) || !client)
+		ready = PLAYER_NOT_READY
+		return FALSE
+
+	if(QDELETED(src) || !client || alert(src,"Are you sure you wish to observe?\nYou will have to wait at least 5 minutes before being able to respawn!","Player Setup","Yes","No") != "Yes")
+		ready = PLAYER_NOT_READY
+		src << browse(null, "window=playersetup") //closes the player setup window
+		new_player_panel()
+		return FALSE
+
+	var/mob/dead/observer/observer = new()
+	spawning = TRUE
+
+	observer.started_as_observer = TRUE
+	close_spawn_windows()
+	var/obj/effect/landmark/observer_start/O = locate(/obj/effect/landmark/observer_start) in GLOB.landmarks_list
+	to_chat(src, "<span class='notice'>Now teleporting.</span>")
+	if (O)
+		observer.forceMove(O.loc)
+	else
+		to_chat(src, "<span class='notice'>Teleporting failed. Ahelp an admin please</span>")
+		stack_trace("There's no freaking observer landmark available on this map or you're making observers before the map is initialised")
+	observer.key = key
+	observer.client = client
+	//observer.set_ghost_appearance()
+	if(observer.client && observer.client.prefs)
+		if(!observer.client.prefs.real_name)
```

Comment:
Some way to handle names with missing prefs would be appreciated as due to other bug there were a few cases with null names as such.

Reply:
>observer.real_name = species.random_name()

Reply:
Yeah but if there's no client or prefs (which is rare) it will fuck up